### PR TITLE
[#101] - Make the quarantine_table argument optional.

### PR DESCRIPTION
**Pull request by danielbeach**
Make the quarantine_table argument optional.
- default `quarantine_table` arg to None
- add `if` logic to opt-in and out of quarantine_table flow.

### CODE REVIEW

File: mack/__init__.py [hunk 1]
```diff
@@ -651,7 +651,7 @@ def constraint_append(
     if not isinstance(append_df, DataFrame):
         raise TypeError("You must provide a DataFrame that is to be appended.")
 
-    if not isinstance(quarantine_table, DeltaTable):
+    if not isinstance(quarantine_table, DeltaTable) and quarantine_table is not None:
```
Comment (jaceklaskowski) [inline mack/__init__.py:654]: Shouldn't `is not None` be before `not isinstance`?

File: mack/__init__.py [hunk 2]
```diff
@@ -629,7 +629,7 @@ def latest_version(delta_table: DeltaTable) -> float:
 
 
 def constraint_append(
-    delta_table: DeltaTable, append_df: DataFrame, quarantine_table: DeltaTable
+    delta_table: DeltaTable, append_df: DataFrame, quarantine_table: DeltaTable = None
```
Comment (MrPowers) [inline mack/__init__.py:632]: I actually think we should make the `quarantine_table` a required argument without a default value.  That way the user needs to explicitly instantiate the function with `quarantine_table=None` if they want to throw out the failing-constraint records.  It seems like a safer user interface.

**Comment by MrPowers** [comment]
Fixes #101

File: mack/__init__.py [hunk 3]
```diff
@@ -665,23 +665,22 @@ def constraint_append(
...
 
     target_details = delta_table.detail().select("location").collect()[0]
-    quarantine_details = quarantine_table.detail().select("location").collect()[0]
+    if quarantine_table is not None:
+        quarantine_details = quarantine_table.detail().select("location").collect()[0]
+        quarantine_df = append_df.filter(
+            "not (" + " and ".join([c for c in constraints]) + ")"
+        )
+        (
+            quarantine_df.write.format("delta")
```
[the remark this answers is not in the document]
Comment (danielbeach) [reply mack/__init__.py:674]: @jaceklaskowski fixed, thanks.

File: mack/__init__.py (hunk 1 above)
Comment (danielbeach) [reply mack/__init__.py]: fixed thanks.

**Comment by MrPowers** [comment]
Thanks for adding this @danielbeach!!

**Pull request merged**